### PR TITLE
Backend: Implemented Progressive FarePolicy PerExtraKm Sections

### PR DESCRIPTION
**Pull request by DonDokidon**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

## Description

1. Refactor  FarePolicyExtraFeeBounds
2. Implemented FarePolicyProgressiveDetailsPerExtraKmSections
	Basically each range of extraKm will have its own perExtraKmRate to cound final fare.

### Additional Changes

- [x] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?

manually

## Checklist

- [x] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Estimate.hs [hunk 1]
```diff
@@ -117,10 +119,5 @@ mkAdditionalBreakups mkPrice mkBreakupItem distance farePolicy = do
       DFP.ProgressiveDetails det -> mkAdditionalProgressiveBreakups det
       DFP.SlabsDetails det -> mkAdditionalSlabBreakups $ DFP.findFPSlabsDetailsSlabByDistance distance det.slabs
 
-    mkAdditionalProgressiveBreakups det = do
```
Comment (hkmangla) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Estimate.hs:120]: why did you remove this?

**Review by hkmangla — APPROVED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Estimate.hs (hunk 1 above)
Comment (DonDokidon) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Estimate.hs:120]: > why did you remove this?

I can't put anything in EXTRA_PER_KM_FARE, it's not constant now, right?

Comment (hkmangla) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Estimate.hs:120]: Yup. For now, just put the startDistance `0` value in this. This param is needed for backward compatibility. 
Later, we can think of creating new param for whole list of per km fares.

**Review by hkmangla — CHANGES_REQUESTED** [review]
